Added : Backed enum to LabelField (#102)
leapt/core-bundle: docs/data_lists/field/label_field.md

> 

Reviewer: ```suggestion
# Label field
```

> ??? usage "Usage"
> ```php
>         $builder->addField('visibility',LabelFieldType::class, [ //Example with a simple text
>                 'label' => 'Visibility',
>                 'mappings'=> [
>                      'everyone' => [
>                         'label' => 'Everyone',
>                         'attr' => ['class' => 'badge badge-succes']
>                     ],
>                     'author' => [
>                         'label' => 'Author and Admin',
>                         'attr' => ['class' => 'badge badge-warning']
>                     ],
>                     'admin' => [
>                         'label' => 'Admin',
>                         'attr' => ['class' => 'badge badge-danger']
>                     ],
>                  ],
>         ])
>             ->addField('status',LabelFieldType::class, [ //Example with an Enum
>                 'label' => 'Statut',
>                 'mappings'=> [
>                     Status::Draft->value => [
>                         'label' => 'Draft publication',
>                         'attr' => ['class' => 'badge badge-warning']                    ],
>                     Status::Published->value => [
>                         'label' => 'Published',
>                         'attr' => ['class' => 'badge badge-succes']                     ],
>                 ],
>         ])
>         ->getDatalist();

Reviewer: ```suggestion
!!! example "Usage"

    ```php
    $builder
        // Example with strings
        ->addField('visibility', LabelFieldType::class, [
            'mappings' => [
                 'everyone' => [
                    'label' => 'Everyone',
                    'attr' => ['class' => 'badge bg-success']
                ],
                'author' => [
                    'label' => 'Author',
                    'attr' => ['class' => 'badge bg-warning']
                ],
                'admin' => [
                    'label' => 'Admin',
                    'attr' => ['class' => 'badge bg-danger']
                ],
             ],
        ])
        // Example with BackedEnum
        ->addField('status', LabelFieldType::class, [
            'mappings' => [
                Status::Draft->value => [
                    'label' => 'Draft publication',
                    'attr' => ['class' => 'badge bg-warning'],
                ],
                Status::Published->value => [
                    'label' => 'Published',
                    'attr' => ['class' => 'badge bg-success'],
                ],
            ],
        ])
        ->getDatalist();
    ```
```

```

> As shown before, you can either use use string, BackedEnum or boolean type as entry.

Reviewer: ```suggestion

As shown above, you can either use scalars or BackedEnums.
```

!!! info "Options"

    Mandatory options:
    

>     * `mappings` : Array that is used to maps the value of the field with the showed value (must be string), the attr is an option of the mappings where you can define class attributes for example.

Reviewer: ```suggestion
    * `mappings`: Array that is used to map the value of the field with the displayed value (must be string). The `attr` key is an option of the mappings where you can define e.g. the HTML class attribute.
```

> [Go back to Data lists documentation](../data_lists.md)

Reviewer: ```suggestion
[Go back to Data lists documentation](../../data_lists.md)
```
